### ISSUE TITLE
Adding a "new constant"?!

### ISSUE THREAD

**Issue by Sonatine1** [PixelRick/CyberpunkSaveEditor#39]
Hi!

The new version loads just fine and most things I know seems to work. I have one problem, though.

When I choose a new item (weapon or whatever), I can't add a "new constant" under "Stats" in the "item data" menu. With existing weapons, clothing etc. which I already edited with a old version (same savegame), it works (it just doesn't shows my old edits anymore... it only shows one totally empty constant).

Any new things I have to consider or am I just too tired to remember that I have to do some steps before?

Best regards
Sona

**New stuff:**
![new](https://user-images.githubusercontent.com/80014836/155836624-71275629-3b4b-487a-ab69-227d273219b7.jpg)

**Already edited stuff:**
![old](https://user-images.githubusercontent.com/80014836/155836629-6229f5ca-399f-4efe-a3ad-65b6d8a350ba.jpg)

**Comment by PixelRick** [comment]
I checked a 1.5 save, and I cannot find any item with stats.. do you ?
If not then it's really deprecated now and better to search for their new way of storing stats..

**Comment by Sonatine1** [comment]
Hey...

OLD Savegames (from 1.1 to 1.5 updated all the time):
- on pre-edited Items I can add constants (the old changes still work ingame, but are NOT shown as you can see on the pic above)
- on new Items I can't do anything
- when I add new constants to pre-added items, the changes work ingame, but when I reload the too, they disappear again

NEW Savegame (1.5):
- on any item I have I can add constants
- these changes work ingame
- thes added constants stay visible in the tool

I will edit here if I find out something new.

**Comment by PixelRick** [comment]
Can you send your save with the Preset_Dian_Yinglong that has previous changes still active but no longer visible in the editor ?
When you say the changes still work, do they after you reload the updated to 1.5 save ?

**Comment by Sonatine1** [comment]
Hey... I'm currently in the train. When I'm home I'll send you everything.

**Comment by Sonatine1** [comment]
Finally home.

The old Savegame (it's my test savegame) has been updated to 1.5. I used your tool before 1.3 and it worked fine. Back then it showed me a whole list of the changes I made (constants added). Now ONLY the weapons I already edited back in early 2021 can be changed with "new constant". When I add a new gun via CET console, it doesn't show me the option "new constant".

And then there is the NEW Savegame, I created a few days ago (because I want to play the game again after 8 months). Here I can choose any gun or item I want and add new constants PLUS (!) these changes are shown afterwards and I have a list of the added constants - on the old savegame, the editable items actually take over these changes in the game itself as well, but it DOESN'T show the new constants afterwards. The list stays empty (only shows them in the running session).

**Old Savegame (basically my test char):**
https://ufile.io/zc2t0e4k

**New Savegame:**
https://ufile.io/ftavzyad

**Comment by PixelRick** [comment]
Thanks ! Made some progress.
I see they now use "modifiersBuffer", it is a DataBuffer with basically looks like a serialized statModifiers array.
Probably to defer load it...
I don't know which one has precedence over the other yet, but I can find out with forged stats.

**Comment by PixelRick** [comment]
![1060_1500_stats_comparison](https://user-images.githubusercontent.com/5144946/155863488-8441b9f0-212a-4bae-8717-85d096071fd7.png)

**Comment by Sonatine1** [comment]
Ah, wow. That's cool! :) So it's basically all the information in one string?!
But they didn't change the file system, right?

Glad that you are able to make some progress. The tool is such a great thing. :)

**Comment by Sonatine1** [comment]
Can you post when you have some Updates? If I can help you, ask me... I can see what I can do. :)

**Comment by PixelRick** [comment]
Yes I will ! It's my second day at work so I have a new coding style and workflow to get used to.
So I'll be probably be too tired to work seriously on it for a few days. This week-end should do.

If you edit a 1.5 save, then load and re-save, does the new save still has your changes visible in the editor ?
(I'd like to know if they move old stats to new stats only when converting on read, or whenever we save.)

**Comment by Sonatine1** [comment]
I'll check this evening and then let you know.

**Comment by Sonatine1** [comment]
Okay. Tested it - both, the old Savegame (updated to 1.5) and the 1.5 Savegame. Reloaded both and saved again.

**Constants are disappearing!** Looks then again like on my previously posted Screenshot named "New stuff".

**Comment by PixelRick** [comment]
Actually that's good, it means i can implement the new buffer and no longer care about the deprecated impl.

**Comment by Sonatine1** [comment]
Ahhh, I'm sorry... I mean it looks like the Screenshot "Already edited stuff".

Jesus... I hope I didn't make you work on a totally false end now. I'm sorry!

**Comment by PixelRick** [comment]
Current pb on my side: I have to add a pre-serialize step just for this object kind to "commit" the edits into the buffer.
However it's hard to hook that in the current rtti implementation. I tried to work on my new rtti impl but I haven't enough free time atm, it will improve once I'm used to the codebase at work.
Seberoth started integrating cybercat in the Wolvenkit. So maybe they'll have the weapon mods editing ready sooner than me.

**Comment by Sonatine1** [comment]
Okay, thanks for the update. :)

**Comment by PixelRick** [comment]
I will probably discontinue this project.
The WolvenKit is gonna do all the same (well in an interpreted language) soon as they have plenty of devs (and some now work at CDPR so they have insider info).
Feels bad to not be able to help them out directly, I hate C# too much.

**Comment by Sonatine1** [comment]
Oh man, I feel sorry for you. Can't they involve you? I mean you spent so much work with this. :/ And will the WolvenKit have the exact same feature?

**Comment by joshuabaker** [comment]
@PixelRick Would love to be able to access this. Is this the “entity editor” you note in the [latest release](https://github.com/PixelRick/CyberpunkSaveEditor/releases/tag/v0.6.0-alpha.1#:~:text=Entity%20editor%20may%20not%20work%20yet%20(not%20tested))?
